src/Vibrato/Vibrato.cpp

```diff
@@ -0,0 +1,135 @@
+#include "Vibrato.h"
+
+VibratoEffector::VibratoEffector(int iSampleRate, int iNumChannel, float fModulationFreqInHz, float fModulationWidthInS)
+    : m_iSampleRate(iSampleRate), 
+      m_iNumChannel(iNumChannel),
+      m_fModulationFreqInHz(fModulationFreqInHz), 
+      m_fModulationWidthInS(fModulationWidthInS) {
+    updateDelayBuffer();
+    updateLfo();
+}
+
+void VibratoEffector::updateDelayBuffer() {
+    deleteBuffer();
+    m_fDelayInSample = m_fModulationWidthInS * static_cast<float>(m_iSampleRate);
+    int iDelayInSample = static_cast<int>(ceil(m_fDelayInSample * 3 + 2));
+
+    int DELAY = static_cast<int>(round(m_fModulationWidthInS * static_cast<float>(m_iSampleRate)));
+    int WIDTH = DELAY;
+    int L = 2 + DELAY + WIDTH * 2;
```

Comment:
Just wondering, if Width and Delay are equated in the previous line, why are both of them used in this formula?

Reply:
I was trying to locate a bug that causes beeps in vibrato output, and I suspect that the vibrato algorithm is incorrect. Therefore I used almost the same code and initialization settings as shown in the reference matlab code, and that's where the equation comes from. Unfortunately that reference code does not provide enough comments on the decision, so I don't know the reason either.

Eventually it turns out that bug comes from logical error in `CRingBuffer::get(float)`. Even though my initial implementation is actually correct and more readable, I just keep this version. Maybe it could be better for reviewers to understand by reverting this file 😂